packages/maker/src/Maker/FrontendApiTypeMaker.php

```diff
@@ -0,0 +1,209 @@
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Input\InputInterface $input
+     * @param \Symfony\Bundle\MakerBundle\ConsoleStyle $io
+     * @param \Symfony\Component\Console\Command\Command $command
+     */
+    #[Override]
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command)
+    {
+        $this->selectedEntityClass = $this->entityChoiceHelper->askForEntity($io, 'What is the entity you want to create the API types for?');
+
+        parent::interact($input, $io, $command);
+    }
+
+    /**
+     * @param \Symfony\Component\Console\Input\InputInterface $input
+     * @param \Symfony\Bundle\MakerBundle\ConsoleStyle $io
+     * @param \Symfony\Bundle\MakerBundle\Generator $generator
+     */
+    #[Override]
+    public function generate(InputInterface $input, ConsoleStyle $io, Generator $generator): void
+    {
+        $fieldsConfig = $this->getFieldsConfig($this->selectedEntityClass);
+
+        $domainClass = $this->findRelatedClass('Domain');
+
+        if ($domainClass !== null) {
+            $fieldsConfig += $this->getFieldsConfig($domainClass);
+        }
+
+        $translationClass = $this->findRelatedClass('Translation');
+
+        if ($translationClass !== null) {
+            $fieldsConfig += $this->getFieldsConfig($translationClass);
+        }
+
+        $entityName = Str::getShortClassName($this->selectedEntityClass);
+        $yamlData = [$entityName => ['type' => 'object', 'config' => ['fields' => [...$fieldsConfig]]]];
+
+        $yaml = Yaml::dump($yamlData, 1000);
+
+        $filename = sprintf('%s/config/graphql/types/ModelType/%s/%s.types.yaml', $this->projectDir, $entityName, $entityName);
+        $this->filesystem->dumpFile($filename, $yaml);
+        $io->writeln(sprintf('File %s created', $filename));
+        $this->writeSuccessMessage($io);
+    }
+
+    /**
+     * @param \Doctrine\ORM\Mapping\ClassMetadata $metadata
+     * @param string $fieldName
+     * @return string
+     */
+    protected function guessApiType(ClassMetadata $metadata, string $fieldName): string
+    {
+        if ($metadata->isSingleValuedAssociation($fieldName)) {
+            $shortClassName = Str::getShortClassName($metadata->getAssociationTargetClass($fieldName));
+            $mapping = $metadata->getAssociationMapping($fieldName);
+            $isNullable = $mapping['joinColumns'][0]['nullable'] ?? false;
+
+            return $shortClassName . ($isNullable ? '' : '!');
+        }
+
+        if ($metadata->isCollectionValuedAssociation($fieldName)) {
+            return '[' . Str::getShortClassName($metadata->getAssociationTargetClass($fieldName)) . '!]!';
+        }
+
+        $apiType = match ($metadata->getTypeOfField($fieldName)) {
+            'guid' => 'Uuid',
+            'integer' => 'Int',
+            'boolean' => 'Boolean',
+            'datetime' => 'DateTime',
+            default => 'String',
```

Comment:
what about money? we already have the separate type when we generate an entity

Reply:
good idea, added in 7e42f90ada